### ISSUE TITLE
2-stage training with sMBR criterion

### ISSUE THREAD

**Issue by PCerles** [SeanNaren/deepspeech.pytorch#403]
Has anyone looked into writing the sMBR loss function? Google uses it in their two-stage training for CTC networks, initializing the network first with CTC loss and then finetuning it with sMBR loss, which more directly optimizes the network for low WER. (http://proceedings.mlr.press/v32/graves14.pdf sec 4).

**Comment by buriy** [comment]
>Google uses it
You're citing 2014 year paper! No recent usages of sMBR in literature.
Maybe rather "Google was once doing it in 2014"?
Also you can check papers here: https://github.com/syhw/wer_are_we

**Comment by PCerles** [comment]
It's mentioned in a subsequent paper from 2015, which is the last character level CTC-based audio paper that their research group has published. They may have switched to a totally different set of architectures but looks like they achieved their best CTC results with this two stage training. The google speech-to-text video API is miles ahead of everyone else from all tests I've done, so s
